lib/sycamore/sycamore/writers/opensearch.py

```diff
@@ -147,7 +131,7 @@ def _string_values_to_python_types(obj: Any):
                     return False
                 elif obj.isnumeric():
                     return int(obj)
-                elif re.match("[0-9]+\.[0-9]+", obj):
+                elif re.fullmatch("-?[0-9]+\.[0-9]+", obj):
```

Comment:
Good catch on -. We both forgot about scientific notation.  The internet suggests 
try: 
  return float(obj)
except ValueError:
  return obj